test/audit_handler_test.go

```diff
@@ -0,0 +1,128 @@
+package test
```

Comment:
unit tests should not be in a different package

and yes, having one `X_test.go` file for each `X.go` source file is a pretty common pattern, unless one file is trivial and doesn't need to be tested in isolation